File: frontend/src/dtos/user/UserProfileDetails.ts
import type { CountryDetails } from "../location/CountryDetails";
import type { UniversitySummary } from "../university/UniversitySummary";
import type { UserStatusSummary } from "./status/UserStatusSummary";

export type UserProfileDetails = {
  userId: number;
  nick: string;
  avatarUrl?: string;
  description: string;
  isFollowed: boolean;
  homeUniversity?: UniversitySummary;
  countryOfOrigin?: CountryDetails;
  status?: UserStatusSummary;
};

File: backend/src/main/java/com/go_exchange_easier/backend/core/domain/user/dto/UserProfileDetails.java
package com.go_exchange_easier.backend.core.domain.user.dto;

import com.go_exchange_easier.backend.core.domain.location.country.CountrySummary;
import com.go_exchange_easier.backend.core.domain.university.dto.UniversitySummary;
import com.go_exchange_easier.backend.core.domain.user.status.UserStatusSummary;

public record UserProfileDetails(

        Integer userId,
        String nick,
        String avatarUrl,
        String description,
        Boolean isFollowed,
        UniversitySummary homeUniversity,
        CountrySummary countryOfOrigin,
        UserStatusSummary status

) { }
